src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -9201,7 +9201,45 @@ public void ParamsArgument_BinaryOperator()
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInArgument, "y").WithArguments("object[]", "object?[]", "y", "object C.operator +(C x, params object?[] y)").WithLocation(4, 45));
         }
 
-        [Fact]
+        [Fact, WorkItem(29954, "https://github.com/dotnet/roslyn/issues/29954")]
+        public void RefLocals_01()
+        {
+            CSharpCompilation c = CreateCompilation(new[] { @"
+class C
+{
+    void Test1(C y1)
+    {
+        ref C x1 = ref y1;
+        y1 = x1;
+    }
+
+    void Test2(C y2)
+    {
+        ref C? x2 = ref y2;
```

Comment:
>ref C? x2 = ref y2; [](start = 8, length = 19)

It feels like there should be a warning here too. It would be possible to assign null to ```y2``` through ```x2```.